src/rpc/misc.cpp

```diff
@@ -158,8 +158,10 @@ UniValue validateaddress(const JSONRPCRequest& request)
 
 #ifdef ENABLE_WALLET
     CWallet * const pwallet = GetWalletForJSONRPCRequest(request);
-
-    LOCK2(cs_main, pwallet ? &pwallet->cs_wallet : nullptr);
+    LOCK(cs_main);
+    if (pwallet) {
+        LOCK(pwallet->cs_wallet);
```

Comment:
This has no effect, the lock is released at the closing brace.

Reply:
Oh, of course! I don't know what I was thinking there :-\  Fixing!